addons/project/tests/test_project_recurrence.py

```diff
@@ -586,13 +598,12 @@ def get_task_and_subtask_counts(domain):
                 task_form.repeat_on_month = 'date'
                 task_form.repeat_day = '15'
                 task_form.date_deadline = datetime(2020, 2, 1)
-                with task_form.child_ids.new() as subtask_form:
-                    subtask_form.name = 'Grandchild task 1 (recurrent)'
-                with task_form.child_ids.new() as subtask_form:
-                    subtask_form.name = 'Grandchild task 2'
-
+                task_form.child_ids.add(grand_child_task_1)
+                task_form.child_ids.add(grand_child_task_2_recurrence)
+                task_form.save()
```

Comment:
Normally the save method is automatically called at the end of the `with` statement.
```suggestion
```